### PR TITLE
Allow max amount of allowed waypoints to be configurable

### PR DESCRIPTION
**Pull request by notzune**
## Description

I realized that you can't configure the max amount of waypoints a player can have which would have been a helpful feature on some servers where there are restrictions on teleportation for whatever reasons they may have.

## Proposed changes

Adds a field in the config.yml that allows to change the max amount of waypoints allowed in the waypoits.yml

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.14.* - 1.19.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java [hunk 1]
```diff
@@ -1,35 +1,7 @@
...
-
-import org.apache.commons.lang.Validate;
-import org.bukkit.Bukkit;
-import org.bukkit.ChatColor;
-import org.bukkit.Location;
-import org.bukkit.NamespacedKey;
-import org.bukkit.OfflinePlayer;
-import org.bukkit.command.CommandSender;
-import org.bukkit.entity.Player;
-import org.bukkit.inventory.ItemStack;
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java:28]: Remove this change

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java [hunk 2]
```diff
@@ -256,7 +242,9 @@ public void addWaypoint(@Nonnull Waypoint waypoint) {
             }
         }
 
-        if (waypoints.size() < 21) {
+        int maxWaypoints = settings.getInt("waypoints.max-allowed");
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java:245]: not a fan of this. We should probably do it under teleporter instead as an item setting

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
This would overflow the menu, wouldn't it...?

**Comment by Sefiraat** [comment]
@notzune Do you intend to keep working on this PR?

**Comment by notzune** [comment]
> @notzune Do you intend to keep working on this PR?

sorry, I made this quick change specifically for a friend and thought that the repo might benefit with my change but I had no intentions to try and make my change better since it worked for my friend and i'm busy with my own things. I greatly apologize

**Comment by Sefiraat** [comment]
> > @notzune Do you intend to keep working on this PR?
> 
> 
> 
> sorry, I made this quick change specifically for a friend and thought that the repo might benefit with my change but I had no intentions to try and make my change better since it worked for my friend and i'm busy with my own things. I greatly apologize

No need to apologise! I will close the PR however if it's not due to be completed. Thanks regardless and you're welcome to reopen or open other PRs whenever, all welcome! <3

**Pull request closed without merging**